hera_cal/nucal.py

```diff
@@ -0,0 +1,473 @@
+        Antenna positions in the form {ant_index: np.array([x,y,z])}.
+    blvec_error_tol : float, default=1e-9
+        Largest allowable euclidean distance a unit baseline vector can be away from an existing
+        cluster to be considered a unique orientation. See "fclusterdata" for more details.
+
+    Returns:
+        Boolean value determining whether or not the baselines are frequency
+        redundant
+
+    """
+    # Split baselines in component antennas
+    _ant1, _ant2, _pol = bl1
+    ant1, ant2, pol = bl2
+
+    # Get baseline vectors
+    blvec1 = antpos[_ant2] - antpos[_ant1]
+    blvec2 = antpos[ant2] - antpos[ant1]
+
+    # Check umode overlap
+    blmag1 = np.linalg.norm(blvec1)
+    blmag2 = np.linalg.norm(blvec2)
+    cond1 = (
+        blmag1 * freqs.min() <= blmag2 * freqs.max()
+        and blmag1 * freqs.max() >= blmag2 * freqs.max()
+    )
+    cond2 = (
+        blmag1 * freqs.min() <= blmag2 * freqs.min()
+        and blmag1 * freqs.max() >= blmag2 * freqs.min()
+    )
+
+    # Check polarization match
+    cond3 = _pol == pol
+
+    # Check headings
+    norm_vec1 = blvec1 / np.linalg.norm(blvec1)
+    norm_vec2 = blvec2 / np.linalg.norm(blvec2)
+    clusters = fclusterdata(
+        np.array([norm_vec1, norm_vec2]), blvec_error_tol, criterion="distance"
+    )
+    cond4 = clusters[0] == clusters[1]
+
+    return (cond1 or cond2) and cond3 and cond4
+
+
+def get_unique_orientations(
+    antpos,
+    reds=None,
+    pols=["nn"],
+    min_ubl_per_orient=1,
+    blvec_error_tol=1e-9,
+    bl_error_tol=1.0,
+):
+    """
+    Sort baselines into groups with the same radial heading. These groups of baselines are potentially
+    frequency redundant in a similar way to redcal.get_reds does. Returns a list of RadialRedundantGroup objects
+
+    Parameters:
+    ----------
+    antpos : dict
+        Antenna positions in the form {ant_index: np.array([x,y,z])}.
+    reds : list of lists
+        List of lists of spatially redundant baselines in the array. Can be found using redcal.get_reds
+    pols : list, default=['nn']
+        A list of polarizations e.g. ['nn', 'ne', 'en', 'ee']
+    min_ubl_per_orient : int, default=1
+        Minimum number of baselines per unique orientation
+    blvec_error_tol : float, default=1e-9
+        Largest allowable euclidean distance a unit baseline vector can be away from an existing
+        cluster to be considered a unique orientation. See "fclusterdata" for more details.
+    bl_error_tol: float, default=1.0
+        The largest allowable difference between baselines in a redundant group
+        (in the same units as antpos). Normally, this is up to 4x the largest antenna position error.
+
+    Returns:
+    -------
+    uors : list of lists of tuples
+        List of list of tuples that are considered to be radially redundant
+    """
+    if reds is None:
+        reds = redcal.get_reds(antpos, pols=pols, bl_error_tol=bl_error_tol)
+    else:
+        reds = deepcopy(reds)
```

Comment:
I was doing this to prevent tuples in reds from being flipped when I'm flipping orientations later in the code. I don't actually think it'll be a problem so I removed it